### PR TITLE
Add redux. Perform real API calls.

### PR DESCRIPTION
**Pull request by varivoda123**

### CODE REVIEW

File: src/sagas/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { all, fork } from 'redux-saga/effects';
+import { watchFetchItem, watchFetchSimilarItems } from './item-saga';
```
Comment (a-bart) [inline src/sagas/index.js:2]: You'd better combine each file sagas into one that will be exported, to reduce the number of sagas in the root file

File: src/sagas/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { watchFetchItems, watchFetchSortedItems } from './items-saga';
```
Comment (a-bart) [inline src/sagas/index.js:3]: the same

File: src/sagas/item-saga.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  }
+}
+
+export function* watchFetchItem() {
+  yield takeLatest(FETCH_ITEM_BEGIN, fetchItem);
+}
+
+export function* watchFetchSimilarItems() {
+  yield takeLatest(FETCH_SIMILAR_ITEMS_BEGIN, fetchSimilarItems);
+}
```
Comment (a-bart) [inline src/sagas/item-saga.js:46]: You could export one saga from a file.
(not necessary, just as advice)

```
export function* watchItemSagas() {
  yield takeLatest(FETCH_ITEM_BEGIN, fetchItem);
  yield takeLatest(FETCH_SIMILAR_ITEMS_BEGIN, fetchSimilarItems);
}
```

File: src/sagas/item-saga.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  fetchSimilarItemsFailure,
+  fetchSimilarItemsBegin,
+} from '../actions/item-actions';
+import API from '../constants/api';
+
+function* fetchItem({ payload }) {
+  const { id } = payload;
+
+  try {
+    const response = yield fetch(`${API.BASE_URL}/${API.MOVIES_ENDPOINT}/${id}`);
```
Comment (a-bart) [inline src/sagas/item-saga.js:17]: Better move all API related logic into a separate file. Handling requests, responses, etc. 

As I see it, a file with api services. e.g.
getItem, getSimilarItems, ...

And more important. When you call some function from saga generator. You need to use saga `call` function
https://redux-saga.js.org/docs/basics/DispatchingActions.html

Screen Shot 2020-06-02 at 10 17 18

File: src/sagas/items-saga.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { put, takeLatest, select } from 'redux-saga/effects';
```
Comment (a-bart) [inline src/sagas/items-saga.js:1]: all points related to prev saga file are applicable for next files

File: src/components/pdp-container/index.jsx [hunk 6]
```diff
@@ -1,26 +1,50 @@
...
-import testItem from '../../data/test-item';
+const PDPContainer = ({ item, itemLoading, similarItems }) => {
+  const dispatch = useDispatch();
+  const initFetch = useCallback(() => {
+    dispatch(fetchItemBegin('447365')); // temporary default value, until routing implemented
+  }, [dispatch]);
+
+  useEffect(() => {
+    initFetch();
+  }, [initFetch]);
```
[the remark this answers is not in the document]
Comment (varivoda123) [reply src/components/pdp-container/index.jsx]: if I remove it form the array everything works fine but I get an eslint error:
image

[the remark this answers is not in the document]
Comment (a-bart) [reply src/components/pdp-container/index.jsx]: I've checked. This is fine, so let's keep it

**Pull request merged**